src/placeos-driver.cr

```diff
@@ -359,24 +361,42 @@ abstract class PlaceOS::Driver
 
             {{method.name.stringify}} => {
               {% for arg in args %}
-                {{arg.name.stringify}} => [
-                  {% if !arg.restriction.is_a?(Union) && arg.restriction.resolve < ::Enum %}
-                    "String",
-                    {% if !arg.default_value.is_a?(Nop) %}
-                      {{arg.default_value}}.to_s
-                    {% end %}
+                {{arg.name.stringify}} => PlaceOS::Driver::Settings.introspect({{arg.restriction.resolve}}).
+                  {% if arg.default_value.is_a?(Nop) %}
+                    merge({ title: {{arg.restriction.resolve.stringify}} }),
                   {% else %}
-                    {{arg.restriction.stringify}},
-                    {% if !arg.default_value.is_a?(Nop) %}
-                      {{arg.default_value}}
-                    {% end %}
+                    merge({ title: {{arg.restriction.resolve.stringify}}, default: {{arg.default_value}} }),
+                  {% end %}
+              {% end %}
+            }{% if args.size == 0 %} of String => Array(String) {% end %},
+          {% end %}
+        }.to_json
+
+        @@functions = funcs = {
+          {% for method in methods %}
+            {% index = 0 %}
+            {% args = [] of Crystal::Macros::Arg %}
+            {% for arg in method.args %}
+              {% if !method.splat_index || index < method.splat_index %}
+                {% args << arg %}
+              {% end %}
+              {% index = index + 1 %}
+            {% end %}
+
+            {{method.name.stringify}} => {
+              {% for arg in args %}
+                {{arg.name.stringify}} => {
+                  {{arg.restriction.stringify}},
+                  {% unless arg.default_value.is_a?(Nop) %}
+                    {{arg.default_value}},
                   {% end %}
-                ],
+                },
               {% end %}
-            }{% if args.size == 0 %} of String => Array(String){% end %},
+            }{% if args.size == 0 %} of String => Array(String) {% end %},
           {% end %}
         }.to_json
-        funcs
+
+        {iface, funcs}
       end
 
       @@security : String?
```

Comment:
This could be a lazy getter
```suggestion
    class_getter security : String do
      Hash(String, Array(String)).new { |h, k| h[k] = [] of String }.tap do |sec|
      {% for method in methods %}
        {% if method.annotation(Security) %}
          level = {{method.annotation(Security)[0]}}.as(::PlaceOS::Driver::Level).to_s.downcase
          sec[level] << {{ method.name.stringify }}
        {% end %}
      {% end %}
      end.to_json
    end
```